sbin/pfmon

```diff
@@ -410,24 +410,22 @@ sub _runtask {
             next;
         }
         my $final_interval = $time_taken > 0 ? $interval - int($time_taken+1) : $interval;
-        if($final_interval < 0) {
-            $final_interval = 0;
-        }
-        # run these functions every $maintenance_interval
         $logger->trace("$id is sleeping for $final_interval ($interval from configuration - $time_taken)");
-        alarm $final_interval;
-        pause;
+        # Only wait if the interval is positive
+        if($final_interval > 0) {
+            alarm $final_interval;
+            pause;
+        }
         last unless $running;
```

Comment:
Move this check inside the if above since that check only matters after doing the pause.